src/YieldDistributor.sol

```diff
@@ -121,6 +131,14 @@ contract YieldDistributor is OwnableUpgradeable {
      * @return uint256 The voting power of the user
      */
     function getCurrentVotingPower(address _account) public view returns (uint256) {
+        /*
+        - This returns the voting power based on the last period, which is intended but may be
+          surprising. It would be good to comment this behavior.
+        - It also uses `cycleLength` (the minimum time between cycles) for this calculation rather
+          than the actual time for the block before the last claimed block. If this is an intentional
+          choice it should be explicitly called out, but I think it might make more sense to track the
+          last two distribution blocks, rather than just `lastClaimedBlockNumber` here.
```

Comment:
I do not understand, the suggestion is to define an interval as a tuple of two blocks, rather than using lastClaimedBlockNumber? What are the benefits in this approach?

Reply:
The only benefit is that it will measure the power accumulated over the previous period between distributions, rather than the `cycleLength` before the previous distribution. I believe that in normal operation, these are expected to be very close. There is only a difference when distributions are not occurring on a `cycleLength` candence.

Consider the following scenario:

`cycleLength == 100`
Block 100: distribution
Block 200: no distribution (because the mechanism is broken or for some other reason)
Block 300: distribution
Block 400: distribution

IIUC, the distribution in block 400 will use the voting power accumulated between blocks 200 and 300. I'm proposing that it may be more fair to use the voting power accumulated between blocks 100 and 300 in this scenario.

Reply:
@bagelface thoughts?

Reply:
I agree with @ShaNagbaImuru on this. `cycleLength` would be more accurately considered the "minimum cycle length" before which a distribution can't happen. Assuming the suggested changes don't have any major tradeoffs, I think it makes for a better algorithm.